Mathlib/Data/List/Basic.lean

```diff
@@ -2915,6 +2919,25 @@ theorem Sublist.map (f : α → β) {l₁ l₂ : List α} (s : l₁ <+ l₂) : m
   filterMap_eq_map f ▸ s.filterMap _
 #align list.sublist.map List.Sublist.map
 
+theorem filterMap_eq_bind_toList (f : α → Option β) (l : List α) :
+    l.filterMap f = l.bind fun a ↦ (f a).toList := by
+  induction' l with a l ih <;> simp
+  rcases f a <;> simp [ih]
+
+theorem filterMap_congr {f g : α → Option β} {l : List α}
+    (h : ∀ x ∈ l, f x = g x) : l.filterMap f = l.filterMap g := by
+  induction' l with a l ih <;> simp
+  simp [ih (fun x hx ↦ h x (List.mem_cons_of_mem a hx))]
+  cases' hfa : f a with b
+  · have : g a = none := Eq.symm (by simpa [hfa] using h a (by simp))
+    simp [this]
+  · have : g a = some b := Eq.symm (by simpa [hfa] using h a (by simp))
+    simp [this]
+
+theorem filterMap_eq_map_of_eq_some {f : α → Option β} {g : α → β} {l : List α}
```

Comment:
```suggestion
theorem filterMap_eq_map_of_forall_eq_some {f : α → Option β} {g : α → β} {l : List α}
```
Is this lemma true as an iff, btw? If so, can you write that iff (and remove the current, weaker, lemma)?

Reply:
ok. I proved the iff, and named it `filterMap_eq_map_iff_forall_eq_some`.